libp2p.nimble

```diff
@@ -19,11 +19,15 @@ requires "nim >= 1.2.0",
          "websock",
          "unittest2 >= 0.0.5 & < 0.1.0"
 
+if (NimMajor, NimMinor) > (1, 2):
+  requires "nimpng#HEAD", "nico"
```

Comment:
That is not correct, these packages are just needed for the tests, they are not a dependency of libp2p itself

Reply:
I will see what we can do. I think we can go with test scope, but I will check if it will work.

Reply:
With current nimble functionality the way to fix the issue is to use taskRequires for each test ask and add the deps like this: 

``` nim
if (NimMajor, NimMinor) > (1, 2):
   taskRequires "test", "nimpng#HEAD", "nico"
   taskRequires "testnative", "nimpng#HEAD", "nico"
   taskRequires "<whatever>", "nimpng#HEAD", "nico"
```

A possible alternative is to expose the action name from here: https://github.com/nim-lang/nimble/blob/master/src/nimblepkg/nimscriptapi.nim#L82

and do something like:

``` nim 
if (NimMajor, NimMinor) > (1, 2) and actionName not in {"build", "install"}:
   requires "nimpng#HEAD", "nico"
```


cc @zah

Reply:
To be more precise, it's only required by `nimble examples_build `